src/ripple/protocol/impl/RippleAddress.cpp

```diff
@@ -556,15 +573,30 @@ void RippleAddress::setAccountPrivate (
     setAccountPrivate (secretKey);
 }
 
-bool RippleAddress::accountPrivateSign (uint256 const& uHash, Blob& vucSig) const
+Blob RippleAddress::accountPrivateSign (Blob const& message) const
 {
-    vucSig = ECDSASign (uHash, getAccountPrivate());
-    const bool ok = !vucSig.empty();
+    if (vchData.size() == 33  &&  vchData[0] == 0xED)
+    {
+        uint8_t const*      secretKey = &vchData[1];
+        ed25519_public_key  publicKey;
+        Blob                signature (sizeof (ed25519_signature));
+
+        ed25519_publickey (secretKey, publicKey);
+
+        ed25519_sign (message.data(), message.size(), secretKey, publicKey, &signature[0]);
+
+        return signature;
+    }
+
+    uint256 const uHash = getSHA512Half (message);
+
+    Blob result = ECDSASign (uHash, getAccountPrivate());
+    bool const ok = !result.empty();
 
     CondLog (!ok, lsWARNING, RippleAddress)
             << "accountPrivateSign: Signing failed.";
 
-    return ok;
+    return result;
```

Comment:
@josh-ripple There's nothing wrong with
```
auto const result = ra.accountPrivateSign(...);
if (result.first)
  foo(result.second);
```

If the private key is bogus then throwing an exception might still be okay. The general public shouldn't be giving private keys to rippled instances. On the other hand if we don't want exceptions we can always go with the pair. Using the `Blob` to communicate error condition is not okay.